Android Source/mobile/src/main/java/net/steppschuh/sensordatalogger/PhoneActivity.java

```diff
@@ -117,6 +129,54 @@ public void onClick(View v) {
             }
         });
 
+        floatingActionRecordButton = (FloatingActionButton) findViewById(R.id.recordActionButton);
+        floatingActionRecordButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                // request runtime permission
+                if (ContextCompat.checkSelfPermission(getApplicationContext(),
+                        Manifest.permission.WRITE_EXTERNAL_STORAGE)
+                        != PackageManager.PERMISSION_GRANTED) {
+                    if (ActivityCompat.shouldShowRequestPermissionRationale(app.getContextActivity(),
+                            Manifest.permission.WRITE_EXTERNAL_STORAGE)) {
+                    } else {
+                        ActivityCompat.requestPermissions(app.getContextActivity(),
```

Comment:
You're currently in the `PhoneActivity`, so every time you need a `Context` you can simply use `this` instead of `getApplicationContext()` or `app.getContextActivity()`.

Reply:
This did not work for me as `this` points to the current instance of `OnClickListener`.

![grafik](https://user-images.githubusercontent.com/24356915/108526864-ede6f680-72d1-11eb-8278-ea8bb98c441b.png)